tests/utils/devServer.js

```diff
@@ -0,0 +1,92 @@
+const cliPath = require('./cliPath')
+const path = require('path')
+const getPort = require('get-port')
+const seedrandom = require('seedrandom')
+const execa = require('execa')
+const pidtree = require('pidtree')
+
+// each process gets a starting port based on the pid
+const rng = seedrandom(`${process.pid}`)
+function getRandomPortStart(rng) {
+  const startPort = Math.floor(rng() * 10000) + 10000 // 10000 to avoid collisions with frameworks ports
+  return startPort
+}
+
+let currentPort = getRandomPortStart(rng)
+
+const startServer = async ({ cwd, env = {} }) => {
+  const tryPort = currentPort++
+  const port = await getPort({ port: tryPort })
+  const host = 'localhost'
+  const url = `http://${host}:${port}`
+  console.log(`Starting dev server on port: ${port} in directory ${path.basename(cwd)}`)
+  const ps = execa(cliPath, ['dev', '-p', port], {
+    cwd,
+    env: { ...process.env, ...env },
+    reject: false,
+  })
+  return new Promise((resolve, reject) => {
+    ps.stdout.on('data', data => {
+      if (data.toString().includes('Server now ready on')) {
+        resolve({
+          url,
+          host,
+          port,
+          close: async () => {
+            const pids = await pidtree(ps.pid).catch(() => [])
+            pids.forEach(pid => () => {
+              try {
+                process.kill(pid)
+              } catch (e) {
```

Comment:
`process.kill()` never throws (I had to look into the Node.js internal code once :) ). In particular, it does not throw when the PID does not exist or has already been killed.

Reply:
I actually had it without the `try/catch` initially but added it later since it did throw.
Also https://github.com/netlify/cli/issues/684 and https://github.com/sindresorhus/execa/issues/96#issuecomment-612787831

Reply:
Oh, sorry, my bad! :+1: